src/sqlfluff/rules/L003.py

```diff
@@ -244,28 +348,23 @@ def _eval(self, segment, raw_stack, memory, **kwargs):
                 "comment_lines": [],
             }
 
-        if segment.is_type("newline"):
```

Comment:
I moved this section of code (with some changes) to a new function `_find_trigger()`.